cmd/xl-storage.go

```diff
@@ -1009,7 +1010,7 @@ func (s *xlStorage) DeleteVersion(ctx context.Context, volume, path string, fi F
 	}
 
 	// when data-dir is specified.
-	if dataDir != "" {
+	if dataDir != "" && fi.TransitionStatus != lifecycle.TransitionPending {
```

Comment:
```suggestion
	if dataDir != "" && fi.TransitionStatus != lifecycle.TransitionPending {
```

Need to add some context here what this means